docs/source/control_tutorials.rst

```diff
@@ -355,26 +364,169 @@ As with :py:class:`.MachineController`, :py:class:`.BMPController` supports the
+::
+
+    >>> import socket
+    >>> from rig.machine_control.consts import SCP_PORT
+    >>> out_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    >>> out_sock.connect((hostname, SCP_PORT))
 
-* :py:class:`~.packets.SDPPacket`
-* :py:class:`~.packets.SCPPacket`
+With the port opened, you can use the
+:py:class:`rig.machine_control.packets.SDPPacket` and
+:py:class:`rig.machine_control.packets.SCPPacket` classes to pack your data
+into properly formatted SDP or SCP packets. Since the ``sark`` and
+``spin1_api`` (unfortunately) make packing/unpacking SDP packets rather clumsy,
+it is common to send and receive SCP packets to applications. 
 
-A blocking implementation of SCP is provided by
-:py:class:`~.scp_connection.SCPConnection`.
+.. note::
 
-These are used internally by :py:class:`.MachineController` and
-:py:class:`.BMPController`. Users are encouraged to read the official SDP and
-SCP App-Notes and refer to the Rig source code for further guidance in using SCP
-and SDP directly in applications.
+    SCP packets are just SDP packets with some additional fields placed in the
+    SDP data payload and when a port number other than 0 are just passed to
+    your application like any other SDP packets.
+
+As an example, to send an SCP packet core 1 on chip (0, 0) with a ``cmd_rc`` of
+``123``::
+
+    >>> from rig.machine_control.packets import SCPPacket
+    >>> data = b"Hello world!\0"
+    >>> packet = SCPPacket(
+    ...     dest_port=1,
+    ...     dest_x=0, dest_y=0, dest_cpu=1,
+    ...     cmd_rc=123
+    ...     data=data
+    ... )
+    >>> out_sock.send(packet.bytestring)
+
+On the receiving core, the ``on_sdp_from_host`` callback might then look like
+this:
+
+.. code-block:: c
+
+    void on_sdp_from_host(uint mailbox, uint port)
+    {
+      sdp_msg_t *msg = (sdp_msg_t *)mailbox;
+      if (msg->cmd_rc == 123)
+      {
+        io_printf(IO_BUF,
+                  "Got SCP packet from host with data: %s\n",
+                  msg->data);
+      }
+      spin1_msg_free(msg);
+    }
 
 .. note::
-    Since different applications typically have very different requirements for
-    SDP and SCP support, Rig does not currently offer any high-level support for
-    their use. The developers are open to discussions about potential
-    (appropriate) high-level interfaces.
+
+    SpiNNaker can only recieve packets up to a certain size. This size can be
+    determined using :py:class:`~rig.machine_control.MachineController`'s
+    :py:meth:`~rig.machine_control.MachineController.scp_data_length` property
+    This property defines the maximum length of the data-field in an SCP packet
+    sent to the machine.
+
+
+Example: Receiving SDP packets from a running application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To receive SDP packets from an application first a socket must be opened
+through which the SDP packets may be received. A port number of your choosing
+may be used for this purpose. For example::
```

Comment:
"To receive SDP packets from an application there must first be an open socket ready to receive the packets."